modules/js-and-css/speculation-rules/hooks.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Hook callbacks used for Speculation Rules.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Prints the speculation rules in a cross-browser compatible way.
+ *
+ * For browsers that do not support speculation rules yet, the rules will not be loaded.
```

Comment:
I think it would be simpler to just always output the `<script type=speculationrules>`, instead of using JavaScript to dynamically insert it. Browsers that do not support `<script type=speculationrules>` will do nothing, which is the same as with the JS-based approach, but with less script execution cost.

Reply:
+1 cf. https://github.com/WordPress/performance/pull/733#discussion_r1204705839